pyatlan/cache/enum_cache.py

```diff
@@ -33,14 +33,17 @@ def refresh_cache(cls) -> None:
         cls.get_cache()._refresh_cache()
 
     @classmethod
-    def get_by_name(cls, name: str) -> Optional[EnumDef]:
+    def get_by_name(cls, name: str) -> EnumDef:
         """
         Retrieve the enumeration definition by its name.
 
-        :param name: human-readable name of the enumeration
-        :returns: the enumeration definition
+        :param name: human-readable name of the enumeration.
+        :raises `NotFoundError`: if the enumeration with the given name does not exist.
+        :returns: enumeration definition
         """
-        return cls.get_cache()._get_by_name(name=name)
+        if not (enum := cls.get_cache()._get_by_name(name=name)):
+            raise ErrorCode.ENUM_NOT_FOUND.exception_with_parameters(name)
```

Comment:
I'm OK with this, as I don't anticipate it'll be a heavily-used area. However, technically this is a breaking change in the interface. Let's just be sure that in the next release notes we indicate this change in behavior for this particular method as a breaking change?

Reply:
Noted!